crates/terminal_view/src/terminal_view.rs

```diff
@@ -117,11 +117,26 @@ impl TerminalView {
         let working_directory =
             get_working_directory(workspace, cx, strategy.working_directory.clone());
 
+        let found_local_worktree = if let Some(working_directory) = working_directory.as_ref() {
```

Comment:
All four large changes in the PR seem to be the same functionality, slightly reworded every time?
Let's create a common method that does all the logic in one place, something like:

```rust
fn create_terminal(
    project: &Model<Project>,
    working_directory: &Path,
    cx: &mut WindowContext,
) -> Result<Model<Terminal>> {
    let local_worktree = project.read(cx).find_local_worktree(working_directory, cx);
    let settings_location = local_worktree
        .as_ref()
        .map(|(worktree, path)| SettingsLocation {
            worktree_id: worktree.read(cx).id().to_usize(),
            path,
        });

    let window = cx.window_handle();
    project.update(cx, |project, cx| {
        project.create_terminal(settings_location, working_directory, None, window, cx)
    })
}
```

it seems that the new function has to be adjusted a bit more to include the `spawn_task` though.

Reply:
I was considering pulling it out into a function but I wasn't sure where to put it or what to name it.

For `spawn_task`, I'm not entirely sure if what I settled on is necessarily the best way to do it. It didn't seem like it's possible to do differently, though.

Reply:
If placing and naming are the only concerns, let's place it in `terminal_view` as the root of the crate and keep the name from my example (or any other one you find more suitable).